examples/transfer-lamports/src/example.ts

```diff
@@ -0,0 +1,197 @@
+     * expired.
+     */
+    rpcSubscriptions,
+});
+
+/**
+ * SETUP: TRANSACTION LIFETIME
+ * Every transaction needs to specify a valid lifetime for it to be accepted for execution on the
+ * network. For this transaction, we will fetch the latest block's hash as proof that this
+ * transaction was prepared close in time to when we tried to execute it. The network will accept
+ * transactions which include this hash until it progresses past the block specified as
+ * `latestBlockhash.lastValidBlockHeight`.
+ *
+ * TIP: It is desirable for your program to fetch this block hash as late as possible before signing
+ * and sending the transaction so as to ensure that it's as 'fresh' as possible.
+ */
+log.info("[setup] Fetching a blockhash for use as the transaction's lifetime constraint");
+const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
+log.info(latestBlockhash, '[setup] Got a blockhash');
+
+/**
+ * STEP 1: Create the transfer transaction
+ */
+
+const transactionMessage = pipe(
+    (log.info('[step 1] Creating a transaction message'), createTransactionMessage({ version: 0 })),
+    /**
+     * Every transaction must state from which account the transaction fee should be debited from,
+     * and that account must sign the transaction. Here, we'll make the source account pay the fee.
+     */
+    tx => (
+        log.info({ address: SOURCE_ACCOUNT_SIGNER.address }, '[step 1] Setting the fee payer'),
+        setTransactionMessageFeePayer(SOURCE_ACCOUNT_SIGNER.address, tx)
+    ),
+    /**
+     * A transaction is valid for execution as long as it includes a valid lifetime constraint. Here
+     * we supply the hash of a recent block. The network will accept this transaction until it
+     * considers that hash to be 'expired' for the purpose of transaction execution.
+     */
+    tx => (
+        log.info(latestBlockhash, '[step 1] Setting the transaction lifetime'),
+        setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, tx)
+    ),
+    /**
+     * Every transaction needs at least one instruction. This instruction describes the transfer.
+     */
+    tx =>
+        appendTransactionMessageInstruction(
+            /**
+             * The system program has the exclusive right to transfer Lamports from one account to
+             * another. Here we use an instruction creator from the `@solana-program/system` package
+             * to create a transfer instruction for the system program.
+             */
+            (log.info(
+                '[step 1] Creating an instruction to transfer Lamports from',
+                SOURCE_ACCOUNT_SIGNER.address,
+                'to',
+                DESTINATION_ACCOUNT_ADDRESS,
+            ),
+            getTransferSolInstruction({
+                amount: lamports(1_000_000n),
+                destination: DESTINATION_ACCOUNT_ADDRESS,
+                source: SOURCE_ACCOUNT_SIGNER,
```

Comment:
I suggest maybe flagging that this is a "signer" type, so that later when you call `signTransactionMessageWithSigners` it's clear how the signature is applied, especially since `signTransactionMessageWithSigners` doesn't take any signers in this example!

Reply:
Thought about that. Will do!